src/commands/emojistats.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::hash_map::HashMap;
+
+use super::*;
+
+use crate::db::emoji_logging::EmojiStats;
+
+#[command]
+#[usage("emojistats [in_text or reactions]")]
+#[usage("emojistats [emoji] ")]
+#[usage("emojistats [emoji] [in_text or reactions]")]
+pub async fn emojistats(ctx: &client::Context, msg: &Message, mut args: Args) -> CommandResult {
+    let db = ctx.get_db().await;
+
+    let single_emoji = match args.single_quoted::<String>().ok() {
+        Some(x) if !crate::util::find_emojis(&x).is_empty() => Some(
+            db.get_emoji_usage(
+                crate::util::find_emojis(&x)
+                    .first()
+                    .user_error(format!("Could not find emoji {}", x).as_str())?,
+            )
+            .await?,
+        ),
+        Some(x) => Some(db.get_emoji_usage_name(&x).await?),
```

Comment:
my review here stays the same: not a fan of the double find_emojis call here, as it's pretty unnecessary.
just do a nested match in the Some(x) case, as was proposed in my review previously